crates/aptos/src/account/derive_resource_account.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::common::types::{CliCommand, CliError, CliTypedResult};
+use aptos_types::account_address::{create_resource_address, AccountAddress};
+use async_trait::async_trait;
+use clap::Parser;
+
+/// A generic interface for allowing for different types of seed phrase inputs
+///
+/// The easiest to use is `string_seed` as it will match directly with the b"string" notation in Move.
+#[derive(Debug, Parser)]
+pub struct ResourceAccountSeed {
+    /// Resource account seed - String BCS encoded
+    ///
+    /// Legacy functionality from previous tooling, this will be renamed.
+    ///
+    /// Seed used in generation of the AccountId of the resource account
+    /// The seed will be converted to bytes using `BCS`
+    #[clap(long, group = "resource_account_seed")]
+    pub(crate) seed: Option<String>,
+
+    /// Resource account seed - String UTF-8 encoded (preferred)
+    ///
+    /// Seed used in generation of the AccountId of the resource account
+    /// The seed will be converted to bytes directly with UTF-8 encoding
+    #[clap(long, group = "resource_account_seed")]
+    pub(crate) string_seed: Option<String>,
```

Comment:
Can we add some docs for examples of these different seeds?

Reply:
Yep, I'm going to fix this anyways, 3 inputs seems crappy.

1 field and 1 encoding field